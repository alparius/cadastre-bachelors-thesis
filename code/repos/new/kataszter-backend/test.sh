#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
