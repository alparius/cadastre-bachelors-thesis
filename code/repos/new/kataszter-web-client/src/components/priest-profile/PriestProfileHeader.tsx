import React, { useState } from "react";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { Button, Confirm, Form, Grid, Header, Icon, Popup } from "semantic-ui-react";

import IPriestProfileData from "../../data/priest/PriestProfileData";
import ProfilePlaceholder from "../ProfilePlaceholder";

interface IPriestProfileHeaderProps {
    priest: IPriestProfileData;
    editMode: boolean;
    isAdmin: boolean;
    handleEditModeChange: (e: any, { value }: any) => void;
    setPriest(newPriest: IPriestProfileData): void;
    asyncDelete(url: string): void;
}

const PriestProfileHeader: React.FC<IPriestProfileHeaderProps> = (props: IPriestProfileHeaderProps) => {
    const { priest, editMode, isAdmin, setPriest, handleEditModeChange, asyncDelete } = props;
    const { formatMessage: fm } = useIntl();

    const [deleteConfirmOpen, setDeleteConfirmOpen] = useState(false);

    const handleDeleteSubmit = () => {
        setDeleteConfirmOpen(false);
        asyncDelete("/priest/" + priest.ID);
    };

    const handleDeleteConfirmOpen = (_: any) => {
        setDeleteConfirmOpen(true);
    };

    const toggleVerified = () => setPriest({ ...priest, verified: !priest.verified });

    if (!priest) {
        return <ProfilePlaceholder />;
    } else {
        return (
            <>
                <Confirm
                    content={fm({ id: "confirm.delete.message.priest" })}
                    open={deleteConfirmOpen}
                    confirmButton={fm({ id: "confirm.delete.confirm" })}
                    onConfirm={handleDeleteSubmit}
                    cancelButton={fm({ id: "confirm.delete.cancel" })}
                    onCancel={() => setDeleteConfirmOpen(false)}
                />

                <Grid columns={12} stackable>
                    <Grid.Row verticalAlign="middle" textAlign="center">
                        <Grid.Column />
                        <Grid.Column width={editMode ? 10 : 7}>
                            <Header as="h1">
                                {priest.name}
                                <span style={{ float: "right" }}>
                                    {editMode && (
                                        <Form.Checkbox
                                            label={fm({ id: "profile.verified.label" })}
                                            name="verified"
                                            toggle
                                            checked={priest.verified}
                                            onChange={toggleVerified}
                                        />
                                    )}
                                    {priest.verified ? (
                                        <Popup
                                            content={fm({ id: "profile.verified.isVerifiedPopup" })}
                                            trigger={<Icon name="check circle outline" fitted color="green" />}
                                        />
                                    ) : (
                                        <Popup
                                            content={fm({ id: "profile.verified.notVerifiedPopup" })}
                                            trigger={<Icon name="question circle outline" fitted color="grey" />}
                                        />
                                    )}
                                </span>
                            </Header>
                        </Grid.Column>
                        <Grid.Column />
                        {!editMode && isAdmin && (
                            <Grid.Column width={3}>
                                <>
                                    <Button
                                        id="editModeButton"
                                        value
                                        onClick={handleEditModeChange}
                                        size="large"
                                        color="violet"
                                        inverted
                                        icon
                                        labelPosition="left"
                                        style={{ marginBottom: "1em", fontFamily: "monospace" }}
                                    >
                                        <Icon name="edit" />
                                        <FM id="profile.edit" />
                                    </Button>
                                    <br />
                                    <Button
                                        id="removeButton"
                                        value
                                        onClick={handleDeleteConfirmOpen}
                                        size="large"
                                        color="red"
                                        inverted
                                        icon
                                        labelPosition="left"
                                        style={{ fontFamily: "monospace" }}
                                    >
                                        <Icon name="trash" />
                                        <FM id="profile.delete" />
                                    </Button>
                                </>
                            </Grid.Column>
                        )}
                    </Grid.Row>
                </Grid>
            </>
        );
    }
};

export default PriestProfileHeader;
