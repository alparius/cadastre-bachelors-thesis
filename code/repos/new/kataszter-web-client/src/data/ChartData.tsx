interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
