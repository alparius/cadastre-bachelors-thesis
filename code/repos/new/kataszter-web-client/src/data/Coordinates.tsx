interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
