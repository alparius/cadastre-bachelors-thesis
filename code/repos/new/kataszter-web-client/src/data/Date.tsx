interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
