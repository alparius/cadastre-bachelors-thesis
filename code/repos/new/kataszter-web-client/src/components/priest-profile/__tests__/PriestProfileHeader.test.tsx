/*===================
  PriestProfileHeader.unit.test.tsx
 ===================*/
import React from "react";

import IPriestProfileData from "../../../data/priest/PriestProfileData";
import { NewTestPriestProfileData } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntl } from "../../../util/test/test-wrapper";
import PriestProfileHeader from "../PriestProfileHeader";

const handlers = {
    handleEditModeChange: jest.fn(),
    asyncDelete: jest.fn()
};

const priest: IPriestProfileData = NewTestPriestProfileData;

describe("PriestProfileHeader element", () => {
    afterEach(() => {
        handlers.handleEditModeChange.mockClear();
    });

    it("should render a priestName when editMode=false and isAdmin=false", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestProfileHeader priest={priest} editMode={false} isAdmin={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a priestName when editMode=true and isAdmin=false", () => {
        const container = createComponentWithIntlAndRouter(<PriestProfileHeader priest={priest} editMode isAdmin={false} {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a priestName and editButton  when editMode=false and isAdmin=true", () => {
        const container = createComponentWithIntlAndRouter(<PriestProfileHeader priest={priest} editMode={false} isAdmin {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a priest name when editMode=true and isAdmin=true", () => {
        const container = createComponentWithIntlAndRouter(<PriestProfileHeader priest={priest} editMode isAdmin {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the handleEditModeChange function when clicked on button", () => {
        const container = mountWithIntl(<PriestProfileHeader priest={priest} editMode={false} isAdmin {...handlers} />);
        container.find("#editModeButton").at(0).simulate("click");
        expect(handlers.handleEditModeChange).toHaveBeenCalledTimes(1);
    });
});
