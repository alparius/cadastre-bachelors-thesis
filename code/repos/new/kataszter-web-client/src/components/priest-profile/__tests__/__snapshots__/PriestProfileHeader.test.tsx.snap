// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestProfileHeader element should render a priest name when editMode=true and isAdmin=true 1`] = `
<div
  className="ui stackable twelve column grid"
>
  <div
    className="center aligned middle aligned row"
  >
    <div
      className="column"
    />
    <div
      className="ten wide column"
    >
      <h1
        className="ui header"
      >
        Janos
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          <div
            className="field"
          >
            <div
              className="ui toggle checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
            >
              <input
                checked={false}
                className="hidden"
                name="verified"
                readOnly={true}
                tabIndex={0}
                type="checkbox"
              />
              <label>
                Verified
              </label>
            </div>
          </div>
          <i
            aria-hidden="true"
            className="grey question circle outline fitted icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </span>
      </h1>
    </div>
    <div
      className="column"
    />
  </div>
</div>
`;

exports[`PriestProfileHeader element should render a priestName and editButton  when editMode=false and isAdmin=true 1`] = `
<div
  className="ui stackable twelve column grid"
>
  <div
    className="center aligned middle aligned row"
  >
    <div
      className="column"
    />
    <div
      className="seven wide column"
    >
      <h1
        className="ui header"
      >
        Janos
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          <i
            aria-hidden="true"
            className="grey question circle outline fitted icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </span>
      </h1>
    </div>
    <div
      className="column"
    />
    <div
      className="three wide column"
    >
      <button
        className="ui violet large icon inverted left labeled button"
        id="editModeButton"
        onClick={[Function]}
        style={
          Object {
            "fontFamily": "monospace",
            "marginBottom": "1em",
          }
        }
        value={true}
      >
        <i
          aria-hidden="true"
          className="edit icon"
          onClick={[Function]}
        />
        Modify
      </button>
      <br />
      <button
        className="ui red large icon inverted left labeled button"
        id="removeButton"
        onClick={[Function]}
        style={
          Object {
            "fontFamily": "monospace",
          }
        }
        value={true}
      >
        <i
          aria-hidden="true"
          className="trash icon"
          onClick={[Function]}
        />
        Remove
      </button>
    </div>
  </div>
</div>
`;

exports[`PriestProfileHeader element should render a priestName when editMode=false and isAdmin=false 1`] = `
<div
  className="ui stackable twelve column grid"
>
  <div
    className="center aligned middle aligned row"
  >
    <div
      className="column"
    />
    <div
      className="seven wide column"
    >
      <h1
        className="ui header"
      >
        Janos
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          <i
            aria-hidden="true"
            className="grey question circle outline fitted icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </span>
      </h1>
    </div>
    <div
      className="column"
    />
  </div>
</div>
`;

exports[`PriestProfileHeader element should render a priestName when editMode=true and isAdmin=false 1`] = `
<div
  className="ui stackable twelve column grid"
>
  <div
    className="center aligned middle aligned row"
  >
    <div
      className="column"
    />
    <div
      className="ten wide column"
    >
      <h1
        className="ui header"
      >
        Janos
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          <div
            className="field"
          >
            <div
              className="ui toggle checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
            >
              <input
                checked={false}
                className="hidden"
                name="verified"
                readOnly={true}
                tabIndex={0}
                type="checkbox"
              />
              <label>
                Verified
              </label>
            </div>
          </div>
          <i
            aria-hidden="true"
            className="grey question circle outline fitted icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </span>
      </h1>
    </div>
    <div
      className="column"
    />
  </div>
</div>
`;
